cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -26,22 +28,25 @@
  * generate the configuration file, it is using the PrintWriter.
  */
 public class KafkaConnectConfigurationBuilder {
-    // Names of environment variables expanded through config providers inside Connect node
-    private final static String PLACEHOLDER_CERT_STORE_PASSWORD_CONFIG_PROVIDER_ENV_VAR = "${strimzienv:CERTS_STORE_PASSWORD}";
     private final static String PLACEHOLDER_SASL_USERNAME_CONFIG_PROVIDER_ENV_VAR = "${strimzienv:KAFKA_CONNECT_SASL_USERNAME}";
     // the SASL password file template includes: <volume_mount>/<secret_name>/<password_file>
     private static final String PLACEHOLDER_SASL_PASSWORD_FILE_TEMPLATE_CONFIG_PROVIDER_DIR = "${strimzidir:%s%s:%s}";
+    // the secrets file template: <namespace>/<secret_name>:<secret_key>
+    private static final String PLACEHOLDER_SECRET_TEMPLATE_KUBE_CONFIG_PROVIDER = "${strimzisecrets:%s/%s:%s}";
 
+    private final Reconciliation reconciliation;
     private final StringWriter stringWriter = new StringWriter();
     private final PrintWriter writer = new PrintWriter(stringWriter);
     private String securityProtocol = "PLAINTEXT";
 
     /**
      * Connect configuration template constructor
      *
-     * @param bootstrapServers  Kafka cluster bootstrap servers to connect to
+     * @param reconciliation the reconciliation
+     * @param bootstrapServers Kafka cluster bootstrap servers to connect to
      */
-    public KafkaConnectConfigurationBuilder(String bootstrapServers) {
+    public KafkaConnectConfigurationBuilder(Reconciliation reconciliation, String bootstrapServers) {
```

Comment:
We only need namespace from the Reconciliation object, which is already available as KafkaConnectCluster class variable from the AbstractModel, so we could directly pass that.

Reply:
I think we have this pattern in all the other builders. Passing just the `Reconciliation` object narrows the scope, while with the entire `KafkaConnectCluster` object (if this is what you mean) you are potentially giving access to more stuff we don't need here.

Reply:
I have followed the pattern in the other builders but I think passing just the namespace would work here too.

Reply:
I would stick with what we have elsewhere so passing reconciliation object.

Reply:
> with the entire KafkaConnectCluster object (if this is what you mean)

This is not what I meant. I was suggesting to *only* use namespace. Anyway, it's ok to pass reconciliation for consistency.